aopy/data.py

```diff
@@ -1069,3 +1074,34 @@ def yaml_read(filename):
         task_codes = yaml.load(file, Loader=yaml.FullLoader)
 
     return task_codes
+
+
+def get_e3v_video_frame_data( digital_data, sync_channel_idx, trigger_channel_idx, samplerate ):
+    
+    """get_e3v_video_frame_data
+
+    Compute pulse times and duty cycles from e3vision video data frames collected on an ecube digital panel.
+
+    Args:
+        digital_data (np.array): [n_time x n_channel] array of data read from ecube digital panel
+        sync_channel_idx (int): sync channel to read from digital_data. Indicates each video frame.
+        trigger_channel_idx (int): trigger channel to read from digital_data. Indicates start/end video triggers.
+        sample_rate (numeric): data sampling rate (Hz)
+
+    Returns:
+        pulse_times (np.array): array of floats indicating pulse start times
+        duty_cycle (np.array): array of floats indicating pulse duty cycle (quotient of pulse width and pulse period)
+    """
+
+    trig_pulse_edges = get_pulse_edge_times(digital_data[:,trigger_channel_idx],samplerate)
+    trig_pulse_times = trig_pulse_edges[:,0]
+    start_trig_time = trig_pulse_times[0]
+    end_trig_time = trig_pulse_times[3]
```

Comment:
a comment here about what the trigger signal looks like might be useful for people looking at this in the future